packages/package.go

```diff
@@ -216,9 +216,44 @@ func getDownloadPath(p Package, t string) string {
 	return path.Join("/epr", p.Name, p.Name+"-"+p.Version+".zip")
 }
 
+// MustParsePackage creates a new package instances based on the given base path.
+// The path passed goes to the root of the package where the manifest.yml is.
+// It runs more strict validation than NewPackage, e.g. ensuring that all categories are valid.
+func MustParsePackage(logger *zap.Logger, basePath string, fsBuilder FileSystemBuilder) (*Package, error) {
+	p, err := newPackage(logger, basePath, fsBuilder)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create package from path %s: %w", basePath, err)
+	}
+	for _, c := range p.Categories {
+		if _, ok := Categories[c]; !ok {
+			return nil, fmt.Errorf("invalid category: %s", c)
+		}
+	}
+	return p, nil
+}
+
 // NewPackage creates a new package instances based on the given base path.
 // The path passed goes to the root of the package where the manifest.yml is.
 func NewPackage(logger *zap.Logger, basePath string, fsBuilder FileSystemBuilder) (*Package, error) {
+	p, err := newPackage(logger, basePath, fsBuilder)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create package from path %s: %w", basePath, err)
+	}
+	j := 0
+	for i, c := range p.Categories {
+		if _, ok := Categories[c]; !ok {
+			p.logger.Warn("package uses an unknown category, will be ignored",
+				zap.String("category", c))
```

Comment:
Previous code was also including the name and version of the package, is there a reason to don't include it anymore?
```suggestion
			p.logger.Warn("package uses an unknown category, will be ignored",
				zap.String("package", p.Name),
				zap.String("version", p.Version),
				zap.String("category", c))
```

Reply:
Previously, it was required because `p.Name` and `p.Version` were set during the `Unpack` method, and therefore those values were not available at this moment.

Related thread: https://github.com/elastic/package-registry/pull/1297#discussion_r2065974532

Reply:
As the logger has been removed from `Package` struct in https://github.com/elastic/package-registry/pull/1333/commits/7dc1e70b3790efe4d07575fbe07675d5433c2418, I just added back those labels for package name and version in the log message.